sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -556,6 +556,9 @@ object NullPropagation extends Rule[LogicalPlan] {
       // a null literal.
       case e: NullIntolerant if e.children.exists(isNullLiteral) =>
         Literal.create(null, e.dataType)
+
+      case n@Not(expr: IsNull) => IsNotNull(expr.child)
+      case n@Not(expr: IsNotNull) => IsNull(expr.child)
```

Comment:
Why do you handle this in NullPropagation?

Reply:
Also, plz check the code style carefully?

Reply:
> Also, plz check the code style carefully?

Sorry, forget to check this place...

Reply:
> Why do you handle this in NullPropagation?

Don't found other better place, and it's also NULL expression problem. 
Or add a new class?

Reply:
Yea, the @viirya suggestion looks nice to me.